src/sql/workbench/api/node/mainThreadNotebookDocumentsAndEditors.ts

```diff
@@ -363,7 +365,7 @@ export class MainThreadNotebookDocumentsAndEditors extends Disposable implements
 		let input = this._instantiationService.createInstance(NotebookInput, uri.fsPath, uri);
 		input.isTrusted = trusted;
 		input.defaultKernel = options.defaultKernel;
-		input.connectionProfileId = options.connectionId;
+		input.connectionProfile = new ConnectionProfile(this._capabilitiesService, options.connectionProfile);
```

Comment:
@kevcunnane could you also close the connect in dispose. Currently there is bug: https://github.com/Microsoft/azuredatastudio/issues/4383? we need close this connect properly.

Reply:
Yurong - can you clarify? This doesn't create a connection, it defines the profile to use when connecting.

Reply:
I asked Matt about QueryEditor, which makes it owns connection and close its own connection. While for SQL NB, we don't make such connection. From this PR title, I thought you are going to make connection for SQL NB.

Reply:
I didn't dig into the detail on how QueryEditor works.